kinto/core/resource/__init__.py

```diff
@@ -717,7 +717,8 @@ def _raise_400_if_invalid_id(self, record_id):
 
         :raises: :class:`pyramid.httpexceptions.HTTPBadRequest`
         """
-        if not self.model.id_generator.match(six.text_type(record_id)):
+        is_string = isinstance(record_id, six.text_type)
```

Comment:
Hum, I don't like that implicit stuff. Should we document it? And then what about booleans, arrays...

IMO if an integer ends up here in the id field, it's a client mistake, that can easily be fixed, no?